### ISSUE TITLE
Cannot install clojure layer

### ISSUE THREAD

**Issue by InvisibleTech** [syl20bnr/spacemacs#7823]
#### Description
After using the **fed** command to add the **clojure** layer to the dot spacemacs file and restarting emacs, I get a compiler error.  After this restart, I try to open a clojure file and there is no REPL menu available when I use SPACE-m.

#### Reproduction guide
- Start Emacs
- use fed
- add **clojure** layer name
- restart emacs
- after all compile messages and loading stop, use ft and navigate to clojure  project and open file.

*Observed behaviour:*
- errors are reported to when compiling the rule.
- after opening clj file, use SPACE-m to open meta menu there are no clojure optios.

*Expected behaviour:*
It is expected that the layer install without errors and that upon opening any clojure file that the SPACE-m menu will show the REPL and other clojure options. 

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: master (rev. 0283f64)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((clojure :variables clojure-enable-fancify-symbols nil)
 auto-completion emacs-lisp git markdown syntax-checking company-mode erlang elixir osx html org colors editorconfig themes-megapack perspectives shell scala go)
```

#### Backtrace
```
In inflection-singularize-string:
inflections.el:138:79:Warning: function `member*' from cl package called at
    runtime

In inflection-pluralize-string:
inflections.el:152:79:Warning: function `member*' from cl package called at
    runtime
^L
Compiling no file at Wed Nov 23 21:38:27 2016
Leaving directory `/Users/name/.emacs.d/elpa/clj-refactor-20161005.344'
^L
Compiling file /Users/name/.emacs.d/elpa/clj-refactor-20161005.344/clj-refactor.el at Wed Nov 23 21:38:27 2016
Entering directory `/Users/name/.emacs.d/elpa/clj-refactor-20161005.344/'
clj-refactor.el:38:1:Error: Symbol's function definition is void: if-let
^L
Compiling no file at Wed Nov 23 21:38:27 2016
```

**Comment by syohex** [comment]
This is caused by https://github.com/clojure-emacs/cider/issues/1882. This is cider issue. However you can fix this issue by updating Emacs version(25.1 or development version).

**Comment by InvisibleTech** [comment]
I am trying that now, thanks
